close_words.py

```diff
@@ -0,0 +1,181 @@
+        print(item, "(", percentage_value, "%) \n")
+    return 0
+
+
+def desempatar(lista):
+    escolha = random.choice(lista)
+    return escolha
+
+
+def print_most_commons(lista1, lista2, word):
+    if len(lista1) > 1 and len(lista2) > 1:
+        immediate_previous = desempatar(lista1)
+        immediate_subsequent = desempatar(lista2)
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    elif len(lista1) > 1 >= len(lista2):
+        immediate_previous = desempatar(lista1)
+        immediate_subsequent = lista2[0]
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    elif len(lista1) <= 1 < len(lista2):
+        immediate_previous = lista1[0]
+        immediate_subsequent = desempatar(lista2)
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    else:
+        return f"{lista1[0]} {word} {lista2[0]}"
+
+
+def frequency(txt):
+    words = text_clean(txt).split()
+    word = input("digite a palavra que quer pesquisar: ")
+    lista_de_busca = []
+    previous = {}
+    subsequent = {}
+
+    for i, item in enumerate(text_clean(txt).split()):
+        if item == word:
+            lista_de_busca.append(i)
+
+    if len(lista_de_busca) > 0:
+        # Conta a frequencia das palavras anteriores e posteriores
+        for j, item in enumerate(words):
+            if item == word:
+                previous_word = words[j - 1]
+                subsequent_word = words[j + 1]
+
+                # Se a palavra já estiver no dict das anteriores, adiciona mais 1 ao seu valor
+                if previous_word in previous:
+                    previous[previous_word] += 1
+                # Se a palavra não estiver no dict das anteriores, adiciona ela e coloque seu valor como 1.
+                else:
+                    previous[previous_word] = 1
+
+                # Se a palavra já estiver no dict das posteriores, adiciona mais 1 ao seu valor
+                if subsequent_word in subsequent:
+                    subsequent[subsequent_word] += 1
+                # Se a palavra não estiver no dict das posteriores, adiciona ela e coloque seu valor como 1.
+                else:
+                    subsequent[subsequent_word] = 1
+
+        print("\nPalavras anteriores à que você escolheu e sua frequência de aparições: ")
+        print(previous)
+        print("\nPalavras posteriores à que você escreveu e sua frequência de aparições: ")
+        print(subsequent, "\n")
+
+        print_percentage(previous, subsequent)
+
+        # Encontra a(s) palavra(s) mais frequente(s)
+        previous_word = max(previous, key=lambda k: previous[k])
+        subsequent_word = max(subsequent, key=lambda k: subsequent[k])
+        # print(previous_word)
+        # print(subsequent_word)
+
+        # Verifica se há empate de palavras mais frequentes
+        previous_word_tie = [word for word in previous if previous[word] == previous[previous_word]]
```

Comment:
entende o  que está fazendo aqui e quais as vantagens dessa abordagem?

Reply:
Sim, ele coloca uma word na previous_word_tie, para cada word que estiver no dict previous e se a previous[word], ou seja, a palvra no indice que estiver sendo percorrido em previous, for igual a previous[previous_word], que é o índice da palavra que aparece mais vezes no dict previous

Reply:
A vantagem é que ele forma uma lista com todas as condições necessárias em apenas uma linha